### ISSUE TITLE
panamax misses latest release

### ISSUE THREAD

**Issue by panzerballett** [panamax-rs/panamax#8]
i just noticed that `panamax` does not always get the latest release of all the crates.

i just ran it. the current latest version of the [`structopt`](https://crates.io/crates/structopt) crate should be `0.3.15` - but `panamax` only synced up to `0.3.12`:
```bash
$ ls -1t | head
0.3.12
0.3.11
0.3.10
0.3.9
0.3.8
0.3.7
0.3.6
0.3.5
0.3.4
0.3.1
```

i have the same issue with e.g. [`libcryptsetup-rs`](https://crates.io/crates/libcryptsetup-rs) (mirrored up to `0.2.1`; should be `0.4.1`)  and [openssl](https://crates.io/crates/openssl) (mirrored up to `0.10.28`; should be `0.10.29`).

am i doing something wrong? or may this be related to the rebasing of the git repository? (i run `panamax` daily using a cron job; but i check only occasionally if there was a problem with the git repo).

(maybe i was wrong when i claimed that re-cloning the git repo fixes all problems... my mirror seems pretty unusable by now.)

**Comment by panzerballett** [comment]
as a quick fix i re-initialized the mirror and downloaded everything again. this seems to work fine.

**Comment by k3d3** [comment]
Glad to hear the workaround fixed it! At the moment, the crates.io-index git handling is very overcomplicated (instead of using a checked out repository, it tries to perform all operations on a bare repo) in an attempt to save some space. I plan on simplifying this significantly in #11 however, which hopefully will make this (and potentially other) bugs go away.

**Comment by panzerballett** [comment]
excellent. great news! thank you very much!
will you update the version on crates.io? or should i compile it from the sources here?
and will i have to re-initialize everything for the change from a bare repo to a checked-out repo?

**Comment by k3d3** [comment]
Yep, I'll make sure to update it once that's complete.

When #11 is finished, you shouldn't need to re-initialize everything, but you will probably have to delete the `crates.io-index` directory. I'll try to make sure that's automatically handled, if possible.

**Comment by panzerballett** [comment]
great, thanks a lot!

i upgraded `panamax` yesterday and had it run tonight. so far everything works perfect! i will keep it running each night for some time now and see what happens.

thanks again!

**Comment by k3d3** [comment]
Should be fixed by #19 . If this still happens, feel free to re-open this ticket.
